api/views.py

```diff
@@ -45,6 +51,18 @@ class DomainAddressViewSet(SaveToRequestUser, viewsets.ModelViewSet):
     def get_queryset(self):
         return DomainAddress.objects.filter(user=self.request.user)
 
+    def perform_create(self, serializer):
```

Comment:
I've been trying to keep as much logic as possible in the Model layer rather than the API layer - so we don't have to re-create the same logic in both places, and risk them falling out of sync with each other.

In this case, I think we can avoid this custom validation logic in the API `perform_create` if we instead add `unique_together = ['user', 'address']` to the `DomainAddress` model. We can still use a `try/except` block to raise the proper status code errors though:
```python
        try:
            serializer.save(user=self.request.user)
        except CannotMakeAddressException as cmae:
            raise exceptions.PermissionDenied(cmae.message)
        except ValidationError ve:
            raise ConflictError({'id': domain_address.id, 'full_address': domain_address.full_address})
```